common/plugin.hpp

```diff
@@ -49,6 +49,7 @@ namespace ILLIXR {
 		 * freed memory. Instead, we call plugin->stop manually before destrying anything.
 		 */
 		virtual void stop() { }
+		virtual void internal_stop() {} 
```

Comment:
This should be in `threadloop` but not `plugin`. `plugin` doesn't really have a notion of "internal stop" because it isn't a thread. Classes that inherit `plugin` (like `threadloop`) can add this individually if they want to.

Reply:
Let's also put a "documentation comment" before it saying this is what a `threadloop` should call if it wants to stop itself.